deployment/install-worker.sh

```diff
@@ -1,40 +1,69 @@
 #!/bin/bash
-set +e
 
+##############################################################################
+# Description:
+# This is a helper script to install the worker binaries on your Apache Spark cluster
+#
+# Usage:
+# ./install-script.sh <cloud-provider> <path-to-worker> <local-worker-installation-path>
+#
+# Sample usage:
+# ./install-script.sh 
+#   azure 
+#   abfs://<blobcontainer>@<gen2storageaccount>.dfs.core.windows.net/<path>/Microsoft.Spark.Worker.netcoreapp2.1.linux-x64-0.1.0.tar.gz 
+#   /usr/local/bin
+##############################################################################
+
+set +e
 # Cloud Provider
 CLOUD_PROVIDER=$1
+
 # Path where packaged worker file (tgz) exists.
-WORKER_PATH=$2
+WORKER_PATH_OR_URI=$2
+
 # The path on the executor nodes where Microsoft.Spark.Worker executable is installed.
-DEST_PATH=$3
+LOCAL_WORKER_PATH=$3
+
 # The path where all the dependent libraies are installed so that it doesn't
-# pollute the $DEST_PATH.
-DEST_PATH_BINARIES=$DEST_PATH/microsoft.spark.worker
+# pollute the $LOCAL_WORKER_PATH.
+SPARKDOTNET_ROOT=$LOCAL_WORKER_PATH/spark-dot-net
+
 # Temporary worker file.
 TEMP_WORKER_FILENAME=/tmp/temp_worker.tgz
 
-# Clean up any existing files. 
-sudo rm -f $DEST_PATH/Microsoft.Spark.Worker
-sudo rm -rf $DEST_PATH_BINARIES
+# Extract version
+IFS='-' read -ra BASE_FILENAME <<< "$(basename $WORKER_PATH_OR_URI .tar.gz)"
+VERSION=${BASE_FILENAME[2]}
+
+IFS='.' read -ra VERSION_CHECK <<< "$VERSION"
+[[ ${#VERSION[@]} != 3 ]] || { echo >&2 "Version check does not satisfy. Raise an issue here: https://github.com/dotnet/spark"; exit 1; }
+
+# Path of the final worker binaries (the one we just downloaded and extracted)
+CURRENT_WORKER_PATH=$SPARKDOTNET_ROOT/Microsoft.Spark.Worker-$VERSION
```

Comment:
why is this current and the comment is final?

Reply:
Done